Public/Functions/WebPSScript/Invoke-WebPSScript.ps1

```diff
@@ -14,21 +18,54 @@ function Invoke-WebPSScript
         [Parameter(Mandatory, ValueFromPipeline)]
         [Alias('WebPSScript')]
         [System.Uri]
-        # The URL of the PowerShell Script to execute.  Redirects are not allowed
-        $Uri
+        $Uri,
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [System.Collections.IDictionary]
+        $Headers
     )
 
     $Uri = $Uri -replace '%20',' '
     Write-Verbose $Uri -Verbose
     
-    if (-NOT (Test-WebConnection $Uri))
+    # Pass Headers through to Test-WebConnection to properly test the connection
+    $Params = @{
+        Uri = $Uri
+    }
+
+    if ($null -ne $Headers)
+    {
+        $Params.Add('Headers', $Headers)
+    }
+
+    if (-NOT (Test-WebConnection @Params))
     {
         Return
     }
-    #[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+
     [Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor [Net.SecurityProtocolType]::Tls1
-    $WebClient = New-Object System.Net.WebClient
-    $WebPSCommand = $WebClient.DownloadString("$Uri")
+
+    # Disable progress bar while Invoke-WebRequest is running for a small performance gain
+    $PreviousProgressPreference = $ProgressPreference
+    $ProgressPreference = 'SilentlyContinue'
+
+    $Params = @{
+        Uri             = $Uri
+        Method          = 'GET'
+        ContentType     = "text/plain"
+        UseBasicParsing = $true
+    }
+
+    if ($null -ne $Headers) {
+        Write-Verbose -Message "Invoke-WebPSScript: Added Headers: $($Headers.Keys -join ', ')"
+        $Params.Add('Headers', $Headers)
+    }
+
+    $WebPSCommand = (Invoke-WebRequest @Params).Content
+
+    # Reset progress bar to whatever it was before
+    $ProgressPreference = $PreviousProgressPreference
+
     Invoke-Expression -Command $WebPSCommand
```

Comment:
this should be changed to something either like this
"$([system.Text.Encoding]::UTF8.GetString($WebPSCommand.ToArray()))"
or this
[System.Text.Encoding]::UTF8.GetString($WebPSCommand)